### PR TITLE
Rework `ResourceBounds` struct

### PR DESCRIPTION
**Pull request by Draggu**
## Introduced changes

- Rework `ResourceBounds` to make it more readable

### CODE REVIEW

File: crates/cheatnet/src/runtime_extensions/forge_runtime_extension/cheatcodes/spoof.rs [hunk 1]
```diff
@@ -38,7 +19,7 @@ pub struct TxInfoMock {
     pub transaction_hash: Option<Felt252>,
     pub chain_id: Option<Felt252>,
     pub nonce: Option<Felt252>,
-    pub resource_bounds: Option<ResourceBounds>,
+    pub resource_bounds: Option<Vec<ResourceBounds>>,
```
Comment (Arcticae) [inline crates/cheatnet/src/runtime_extensions/forge_runtime_extension/cheatcodes/spoof.rs:22]: how come this is now vec and we don't need to change anything in cairo? 🤔

Comment (Draggu) [reply crates/cheatnet/src/runtime_extensions/forge_runtime_extension/cheatcodes/spoof.rs:22]: Previously it was `ResourceBounds(pub Vec<Felt252>)` now it's `<Vec<ResourceBounds>>` where 
```rs
struct ResourceBounds {
    pub resource: Felt252,
    pub max_amount: u64,
    pub max_price_per_unit: u128,
}
```

Previously was this way to keep deserialising logic. Current is semantically correct

Comment (piotmag769) [reply crates/cheatnet/src/runtime_extensions/forge_runtime_extension/cheatcodes/spoof.rs:22]: Nothing changed in cairo, we just changed the interface in Rust (btw we aligned it with the cairo one). It still serializes to felts the same way

**Review by piotmag769 — APPROVED** [review]
Rly gj with the changes

**Pull request merged**